core/src/test/java/org/apache/iceberg/TestLocationProvider.java

```diff
@@ -297,4 +297,17 @@ public void testObjectStorageWithinTableLocation() {
     Assert.assertEquals(
         "Fourth part should be the file name passed in", "test.parquet", parts.get(3));
   }
+
+  @Test
```

Comment:
Put up for discussion first while writing tests in case we want to talk about it during community sync